File: examples/kafka-quickstart/src/main/java/acme/Receiver.java
package acme;

import java.time.Instant;
import java.util.concurrent.CompletionStage;

import jakarta.enterprise.context.ApplicationScoped;

import org.apache.kafka.common.header.Headers;
import org.eclipse.microprofile.reactive.messaging.Incoming;

import io.smallrye.reactive.messaging.kafka.KafkaRecord;

@ApplicationScoped
public class Receiver {

    @Incoming("kafka")
    public CompletionStage<Void> consume(KafkaRecord<String, String> message) {
        String payload = message.getPayload();
        String key = message.getKey();
        Headers headers = message.getHeaders();
        int partition = message.getPartition();
        Instant timestamp = message.getTimestamp();
        System.out.println("received: " + payload + " from topic " + message.getTopic());
        return message.ack();
    }

}

File: examples/kafka-quickstart-kotlin/src/main/kotlin/acme/Receiver.kt
package acme

import io.smallrye.reactive.messaging.kafka.KafkaRecord
import org.eclipse.microprofile.reactive.messaging.Incoming
import java.util.concurrent.CompletionStage
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
open class Receiver {

    @Incoming("kafka")
    open fun consume(message: KafkaRecord<String, String>): CompletionStage<Void> {
        val payload = message.payload
        val key = message.key
        val headers = message.headers
        val partition = message.partition
        val timestamp = message.timestamp
        println("received: " + payload + " from topic " + message.topic)
        return message.ack()
    }

}
